### ISSUE TITLE
Bump Brave's Windows updater (Omaha) to v1.3.361.113

### ISSUE THREAD

**Issue by mherrmann** [brave/brave-browser#22060]
## Description 

We want to [disable some statistics collection in Brave's auto-updater on Windows](https://github.com/brave/omaha/issues/41). We're taking this as an opportunity to make a few other small improvements to the auto-updater. Specifically:

 1. [Disable some statistics collection in Brave's auto-updater on Windows](https://github.com/brave/omaha/issues/41).
 2. [Avoid local changes after building Omaha](https://github.com/brave/brave-browser/issues/21103)
 3. [Abort Omaha's build with meaningful messages upon error](https://github.com/brave/omaha/pull/40)
 4. [Omaha's build fails when certain environment variables are not set](https://github.com/brave/brave-browser/issues/21528)

This issue should count as resolved once stats collection no longer occurs as described in https://github.com/brave/omaha/issues/41 and most of Brave's auto-update fleet on Windows uses the new updater version.

**Comment by mherrmann** [comment]
Re-opening as I've had to roll back #12868 in https://github.com/brave/brave-core/pull/12874.

**Comment by LaurenWags** [comment]
@brave/legacy_qa we should check Win 10 and 11 at a minimum. Win 8 and 7 would be helpful, but not strictly necessary.

**Comment by GeetaSarvadnya** [comment]
QA/Blocked until the clarification gets resolved https://github.com/brave/brave-core/pull/12895#issuecomment-1122277252

**Comment by GeetaSarvadnya** [comment]
1. Follow the remove updater as described in the "Cleaning up after a test" in https://github.com/brave/brave-core/pull/11096.
2. Install the `BraveBrowserStandaloneBetaSetup.exe` build version`1.34.60` (build which is before the Omaha updater change `1.3.361.113`)
3. Confirmed build is installed successfully and version details are shown correctly in brave://settings/help
4. Confirmed that the old Omaha updater `1.3.361.111` is shown in the registry as expected
5. Confirmed old Omaha updater version `1.3.361.111` is shown in `C:\Program Files (x86)\BraveSoftware\Update`
6. Go to brave://settings/help and wait for update and click on `Relaunch` and ensured brave is updated to the latest public beta version 1.39.x

@mherrmann Once the build is upgraded to `1.39.x` beta, the Omaha updater version is still `1.3.361.111` in the registry, looks like it's expected as updater is using the old Omaha updater to perform the update. Just wanted to reconfirm is this expected?

**Comment by mherrmann** [comment]
@GeetaSarvadnya it might be OK. Do background updates update the updater to `.113`? Please see "Background updates" in https://github.com/brave/brave-core/pull/11096 for the steps necessary to test this.

**Comment by GeetaSarvadnya** [comment]
@mherrmann: I ran `BraveSoftwareUpdateTaskMachineUA` task multiple times in windows task scheduler to check the brave background updates, but Omaha updater is not updating to the updated version `.113` . The new updater isn't yet uploaded to the update server?

**Comment by mherrmann** [comment]
@GeetaSarvadnya this was my mistake. The new updater is not yet uploaded to the update server (otherwise, all users would already be receiving it).

I've uploaded the new updater to the _staging_ update server. You can configure your machine to pull updates from staging by setting the registry string value `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\BraveSoftware\UpdateDev\url` to `https://updates.brave.com/service/update2`. You need to be connected to VPN to be able to access the staging update server. (To test, you can open https://updates.brave.com. If you get a 404 error, then you have access.)

The above should let you update the updater to `1.3.361.113`. If you encounter any issues, please just let me know.

**Comment by MadhaviSeelam** [comment]
Verification `In-Progress` using
```
Brave | 1.39.101 Chromium: 101.0.4951.64 (Official Build) beta (64-bit)
-- | --
Revision | d1daa9897e1bc1d507d6be8f2346e377e5505905-refs/branch-heads/4951@{#1208}
OS | Windows 11 Version 21H2 (Build 22000.675)
```

- Verified the test plan from https://github.com/brave/brave-core/pull/12895

### Windows 11x64
**_1. Verify installers and auto-updater work on a clean profile_**

<details>
<summary>Case 1: Verify the Omaha updater version when 1.39.x is installed_PASSED</summary>

**Pre-requisites:**
(Please note these steps are same as remove updater as described in the "Cleaning up after a test" in https://github.com/brave/brave-core/pull/11096 but I have breakdown the steps into detailed)

1. Uninstall all the brave version (all channels Dev, Release, Beta and Nightly) via Control panel
2. Remove all the local profile folders from `%LOCALAPPDATA%\BraveSoftware`
3. Go to `C:\Program Files (x86)\BraveSoftware\update` path and observe that `Update` folder is still available and updater version `1.3.361.113` is still shown
4. Open RegEdit and navigate to `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\BraveSoftware\Update` and ensure Omaha updater version is `1.3.361.113` 
5. Open command prompt with Admin privileges and run the command `"C:\Program Files (x86)\BraveSoftware\Update\BraveUpdate.exe" /uninstall`
6. Go to C:\Program Files (x86)\BraveSoftware path and observe that the `update` folder is removed
7. Open RegEdit and navigate to `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\BraveSoftware\Update` and ensure Omaha updater version `1.3.361.113` is removed from the registry
8. If the registry key isn't empty in steps 6 and 7 then kill all instances of `BraveUpdate.exe`, `BraveCrashHandler.exe`, `BraveCrashHandler64.exe` in the Task Manager
9. Steps 6 and 7 indicate that the Omaha updater is completely uninstalled''

<img width="485" alt="registry after cleanup" src="https://user-images.githubusercontent.com/98358127/168401476-732905e2-2bea-492e-ad75-67383c6f2006.png">


**Steps:**
- Run the steps 1 to 7 from Pre-requisites
- Download `BraveBrowserStandaloneBetaSetup.exe` and install the standalone binary 1.39.x with Admin rights and ensured that standalone installer installed successfully
- Confirmed that the new Omaha updater `1.3.361.113` is shown in the registry as expected
- Confirmed new Omaha updater version `1.3.361.113` is shown in `C:\Program Files (x86)\BraveSoftware\Update`
- Uninstall the 1.39.x from control panel and remove the profile folder from the `%LOCALAPPDATA%\BraveSoftware`
- Install the 1.39.x again and confirmed that the brave installed successfully
- Go to brave://settings/help and wait for the update and and click on the `Relaunch` and confirmed that brave is upgraded to latest public version

ex|ex|ex|ex
---|---|---|---
<img width="580" alt="braveupdate" src="https://user-images.githubusercontent.com/98358127/168401325-c6fc515b-ced1-46db-9c38-4965f279d465.png">|<img width="580" alt="Registry updater" src="https://user-images.githubusercontent.com/98358127/168401346-218f0612-09ae-4821-994e-44aa3af046f1.png">|<img width="953" alt="logs" src="https://user-images.githubusercontent.com/98358127/168401385-00da8e5e-e5f9-4d5a-8c47-b7756d81a890.png">|<img width="459" alt="upgrade from 101" src="https://user-images.githubusercontent.com/98358127/168401402-df0d6a0c-22a2-49a4-b1bd-6b9e1c1ab87a.png">

</details>

<details>
<summary>Case 2: Download and install stub installer binary 1.39.x with Admin rights_PASSED</summary>


- Uninstall 1.39.x from the control panel if it's installed
- Remove local profile folders from `%LOCALAPPDATA%\BraveSoftware`
- Install 1.39.x  stub installer `BraveBrowserBetaSetup.exe` and ensured that brave is installed successfully and latest public released beta version is fetched from the Omaha server as expected

<img width="457" alt="Testcase 2-test stub installer -public beta" src="https://user-images.githubusercontent.com/98358127/168401763-0d7a2486-3759-4b0c-9116-7d8b805a696b.png">

</details>

**Comment by mherrmann** [comment]
I took the staging update server `updates.brave.com` back offline because @GeetaSarvadnya said she was done with it.

**Comment by LaurenWags** [comment]
@mherrmann we still have to test this issue with Win 11 (@MadhaviSeelam is doing) and Win 8 (@kjozwiak will do), so sounds like we might need that back.

**Comment by mherrmann** [comment]
@LaurenWags I see. I'm bringing staging back online.

**Comment by LaurenWags** [comment]
Verification in progress with Win 10 x86 using below version:

```
Brave	1.39.106 Chromium: 101.0.4951.67 (Official Build) beta (32-bit) 
Revision	8888ee7a24e2c36661ddb9536c35b7d4852a3a98-refs/branch-heads/4951@{#1230}
OS	Windows 10 Version 21H2 (Build 19044.1682)
```

- Verified the test plan from https://github.com/brave/brave-core/pull/12895

### Windows 10 x64
**_1. Verify installers and auto-updater work on a clean profile_**

<details>
<summary>Case 1: Verify the Omaha updater version when 1.39.x is installed - InProgress</summary>

**Pre-requisites:**
(Please note these steps are same as remove updater as described in the "Cleaning up after a test" in https://github.com/brave/brave-core/pull/11096 but I have breakdown the steps into detailed)

1. Uninstall all the brave version (all channels Dev, Release, Beta and Nightly) via Control panel
2. Remove all the local profile folders from `%LOCALAPPDATA%\BraveSoftware`
3. Go to `C:\Program Files (x86)\BraveSoftware\update` path and observe that `Update` folder is still available and updater version `1.3.361.113` is still shown
4. Open RegEdit and navigate to `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\BraveSoftware\Update` and ensure Omaha updater version is `1.3.361.113` 
5. Open command prompt with Admin privileges and run the command `"C:\Program Files (x86)\BraveSoftware\Update\BraveUpdate.exe" /uninstall`
6. Go to C:\Program Files (x86)\BraveSoftware path and observe that the `update` folder is removed
7. Open RegEdit and navigate to `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\BraveSoftware\Update` and ensure Omaha updater version `1.3.361.113` is removed from the registry
8. If the registry key isn't empty in steps 6 and 7 then kill all instances of `BraveUpdate.exe`, `BraveCrashHandler.exe`, `BraveCrashHandler64.exe` in the Task Manager
9. Steps 6 and 7 indicate that the Omaha updater is completely uninstalled

**Steps:**
1. Run the steps 1 to 7 from Pre-requisites above - N/A as I used a fresh VM with Brave never having been installed
2. Download `BraveBrowserStandaloneBetaSetup32.exe` for 1.39.106 and install the standalone binary 1.39.x with Admin rights and ensured that standalone installer installed successfully
3. Confirmed that the new Omaha updater `1.3.361.113` is shown in the registry as expected
4. Confirmed new Omaha updater version `1.3.361.113` is shown in `C:\Program Files (x86)\BraveSoftware\Update`
5. Uninstall the 1.39.x from control panel and remove the profile folder from the `%LOCALAPPDATA%\BraveSoftware`
6. Install the 1.39.x again and confirmed that the brave installed successfully
7 Go to brave://settings/help and wait for the update and and click on the `Relaunch` and confirmed that brave is upgraded to latest public version

Example | Example | Example | Example
-- | -- | -- | --

 
</details>

**Comment by btlechowski** [comment]
Verification passed on

Brave | 1.39.111 Chromium: 102.0.5005.61 (Official Build) (64-bit)
-- | --
Revision | 0e59bcc00cc4985ce39ad31c150065f159d95ad3-refs/branch-heads/5005@{#819}
OS | Windows 7 Service Pack 1 (Build 7601.24544)


- Verified the test plan from https://github.com/brave/brave-core/pull/12895

### Windows 7 x64
**_1. Verify installers and auto-updater work on a clean profile_**

<details>
<summary>Case 2: Download and install stub installer binary 1.39.x with Admin rights_PASSED</summary>


- Uninstall 1.39.x from the control panel if it's installed
- Remove local profile folders from `%LOCALAPPDATA%\BraveSoftware`
- Install 1.39.x  stub installer `BraveBrowserBetaSetup.exe` and ensured that brave is installed successfully and latest public released beta version is fetched from the Omaha server as expected
![image](https://user-images.githubusercontent.com/34715963/169907320-61d3d25c-ee4e-47a1-aea0-331c53d0a525.png)
![image](https://user-images.githubusercontent.com/34715963/169907309-62232e47-fdd4-43a5-a30e-b3eef35c2370.png)


</details>


**_2. Verify installers and auto-updater works on upgrade profile_**

<details>
<summary>Case 2: Install standalone installer 1.38.x and upgrade the profile to 1.39.x_PASSED</summary>

- Install standalone beta installer 1.38.x
- Go to brave://settings/help and wait for the update and click on `Relaunch`
- Confirmed that brave is upgraded to 1.39.x public released version

</details>


**_3. Verify installers and auto-updater work when the user declines the Admin rights in UAC_**

[**Note** : when brave is installed for the current user (without admin rights) the `Brave-Browser-Beta` folder gets created under the registry path `Computer\HKEY_CURRENT_USER\SOFTWARE\BraveSoftware\` and Omaha new updater version is shown as expected (detailed testing for the current user is covered in https://github.com/brave/brave-browser/issues/22179#issuecomment-1112647799]


<details>
<summary>Case 2: Verify stub installer can be installed without admin rights_PASSED</summary>

- Install 1.39.x beta stub installer
- Decline the Admin rights in UAC while installing the setup
- Confirmed 1.39.x public released beta version is installed

![image](https://user-images.githubusercontent.com/34715963/169907894-0435ae49-aaca-4590-a4da-3cb56be9a9b6.png)
![image](https://user-images.githubusercontent.com/34715963/169907963-8182d215-8f29-4012-907a-0cb176f07d07.png)
![image](https://user-images.githubusercontent.com/34715963/169908322-ff559371-b3d1-4279-88c0-6c75145d72a0.png)


</details>


**_4. Verify re-install same installer version works_**

<details>
<summary>Case 1: Verify standalone installer can be installed twice and update works_PASSED</summary>

- Install 1.39.x standalone installer twice 
- Ensured two brave windows are opened with the same version
- Go to brave://settings/help and wait for the update and click on `Relaunch`
- Confirmed that brave is updated to 1.39.x public released version

</details>

**_5. Install quite older installers and check update works correctly_**

<details>
<summary>Case 1: Install 1.30.x and ensure upgrade works_PASSED</summary>

- Install 1.30.x beta build
- Go to brave://settings/help and wait for update and click on `Relaunch` 
- Confirmed 1.39.x public released version is installed

</details>

**_6. Verify that install brave works when the Internet is disconnected_**

<details>
<summary>Case 1: Verify standalone installer can be installed when the network is disconnected_PASSED</summary>

- Install 1.39.x standalone installer
- Confirmed standalone installer is installed successfully and version details are shown correctly
- Go to brave://settings/help and ensured update is NOT shown which is expected as to pull the Omaha version network needs to be connected

</details>


 **_7. Verify installing a new binary by older binary from different installers (e.g one is standalone and the other is stub/silent)_**
<details>
<summary>Case 1: Install standalone installer 1.38.x and upgrade it by 1.39.x stub installer_PASSED</summary>

- Clean profile 1.38.x standalone installer `BraveBrowserStandaloneSetup.exe`
- Install the standalone installer and confirmed 1.38.x is installed successfully
- Upgrade the profile by 1.39.x stub installer `BraveBrowserBetaSetup.exe`
- Confirmed that the 1.39.x public releases version is installed successfully

</details>


--------
